base/b64/codec.go

```diff
@@ -21,13 +21,12 @@ type decoder struct {
 }
 
 func (d decoder) Decode(v interface{}) error {
```

Comment:
Oh.... treating encodings as codecs feels very wrong to me.  The interface doesnt mesh well, and its a separate step altogether.

Theres a reason that cids have the base encoding as a separate field from the codec